DAQ/src/ArtBinaryPacketsFromDigis_module.cc

```diff
@@ -0,0 +1,1113 @@
+    const size_t number_of_crystals_per_roc = 8;
+    const size_t number_of_calo_rocs_per_dtc = 6;
+
+    int _generateTextFile;
+
+    // Diagnostics level.
+    int _diagLevel;
+
+    int _enableFPGAEmulation;
+
+    // Limit on number of events for which there will be full printout.
+    int _maxFullPrint;
+
+    // Label of the module that made the hits.
+    art::ProductToken<StrawDigiCollection> const  _sdtoken;
+    art::ProductToken<CaloDigiCollection>  const  _cdtoken;
+
+    const Calorimeter* _calorimeter; // cached pointer to the calorimeter geometry
+
+    void   fillEmptyHeaderDataPacket(DataBlockHeader& HeaderData, uint64_t& EventNum, uint8_t& ROCId, uint8_t& DTCId, uint8_t Subsys);
+    void   printHeader(DataBlockHeader const& headerDataBlock);
+
+    //--------------------------------------------------------------------------------
+    //  methods used to process the tracker data
+    //--------------------------------------------------------------------------------
+    void   fillTrackerDataPacket(const StrawDigi& SD, TrackerDataPacket& TrkData);
+
+    void   fillTrackerHeaderDataPacket(const StrawDigi& SD, DataBlockHeader& HeaderData, uint64_t& EventNum);
+
+    void   fillEmptyTrackerDataPacket(TrackerDataPacket& TrkData);
+
+    void   processTrackerData(art::Event& evt, uint64_t& eventNum,
+			      tracker_data_block_list_t &trackerData);
+
+
+    void   fillTrackerDMABlocks(raw_data_list_t& dataStream, tracker_data_block_list_t const& trackerData);
+
+    void   fillTrackerDataStream(raw_data_list_t& dataStream, tracker_data_block_t const& trkData);
+
+    void   printTrackerData(TrackerDataPacket const& curDataBlock);
+
+    //--------------------------------------------------------------------------------
+    //  methods used to handle the calorimeter data
+    //--------------------------------------------------------------------------------
+    void   fillCalorimeterDataPacket(const CaloDigi& SD, CaloDataPacket& caloData);
+    void   addCaloHitToCaloPacket(calo_data_block_t& dataBlock,
+				  CaloDataPacket& caloData);
+
+    void   fillCalorimeterHeaderDataPacket(const CaloDigi& SD, DataBlockHeader& HeaderData, uint64_t& EventNum);
+
+    void   fillEmptyCalorimeterDataPacket(CaloDataPacket& caloData);
+    void   fillHeaderByteAndPacketCounts(calo_data_block_t& caloData);
+
+    void   processCalorimeterData(art::Event& evt, uint64_t& eventNum,
+				  calo_data_block_list_t& caloDataBlocks);
+
+
+    void   fillCalorimeterDMABlocks(raw_data_list_t& dataStream, calo_data_block_list_t& caloData);
+
+    void   fillCalorimeterDataStream(raw_data_list_t& dataStream, calo_data_block_t& caloData);
+
+    void   printCalorimeterData(CaloDataPacket const& curDataBlock);
+
+    size_t waveformMaximumIndex(std::vector<adc_t>& waveform);
+
+    void flushBuffer(raw_data_list_t& buffers);
+    void closeDataBuffer(raw_data_list_t& dataStream, bool openNew = true) {     
+      uint64_t sz = dataStream.back().second;
```

Comment:
Something this complex should not be implemented inside the declaration.  In this case it makes no real difference since, for modules, declaration and implementation are in a single file.  But it's good practice to keep large implementations out of the declarations.